test_data/std/keccakf16_memory_test.asm

```diff
@@ -0,0 +1,117 @@
+use std::machines::hash::keccakf16_memory::Keccakf16Memory;
+use std::machines::small_field::memory::Memory;
+use std::machines::small_field::add_sub::AddSub;
+use std::machines::range::Byte2;
+use std::machines::range::Bit12;
+
+machine Main with degree: 65536 {
+    reg pc[@pc];
+
+    reg X1[<=];
+    reg X2[<=];
+
+    reg Y1[<=];
+    reg Y2[<=];
+
+    Byte2 byte2;
+    Bit12 bit12;
+    AddSub add_sub(byte2);
+    Memory memory(bit12, byte2);
+
+    Keccakf16Memory keccakf16_memory(memory, add_sub);
+
+    col fixed STEP(i) { i };
+
+    // Big endian.
+    // Usage: mstore addr_h, addr_l, val_h, val_l
+    instr mstore X1, X2, Y1, Y2 -> link ~> memory.mstore(X1, X2, STEP, Y1, Y2);
+    // Usage: keccakf16_memory input_addr_h, input_addr_l, output_addr_h, output_addr_l;
+    instr keccakf16_memory X1, X2, Y1, Y2 -> link ~> keccakf16_memory.keccakf16_memory(X1, X2, Y1, Y2, STEP);
+
+    col witness val_h, val_l;
+    // Usage: assert_eq addr_h, addr_l, val_h, val_l
+    instr assert_eq X1, X2, Y1, Y2 ->
+        link ~> (val_h, val_l) = memory.mload(X1, X2, STEP)
+    {
+        val_h = Y1,
+        val_l = Y2
+    }
+
+    function main {
+        // Test 1: Input/output address computations have no carry.
+        // 0 for all 25 64-bit inputs except setting the second 64-bit input to 1. All 64-bit inputs in chunks of 4 16-bit big endian limbs.
```

Comment:
what's the endianness of each 64-bit number?  this comment says the second input is 1, but in the mstores below `1` is set to the 4th limb which should belong to the first input?

Reply:
It should be big endian to match with other machines.

Each `mstore` here actually stores 32 bits (two 16 bit limbs), so it's actually the 4th `mstore` rather than the 4th limb, which corresponds to the least significant limb of the second 64 bit number, so I think it should be correct. The test result also matches the original non memory version.